### PR TITLE
Feature/module 6 redux

### PR DESCRIPTION
**Pull request by UltramarineDev**

### CODE REVIEW

File: netflix-roulette/src/components/actions.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  if (filter) {
+    url = `${url}&filter=${filter}`;
+  }
+  const response = await fetch(url).then(res => res.json());
+  dispatch(getMoviesSuccessAction(response.data));
+};
+
+export const getMovieAction = (id) => async (dispatch) => {
+   const response = await fetch(`${constants.URL}/${id}`).then(res => res.json());
+   dispatch(getMovieSuccessAction(response));
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]: Probably you can dispatch `getMovieSuccessAction` inside `then()`

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js [hunk 2]
```diff
@@ -46,4 +51,23 @@ const MovieDetails = () => {
...
+  getMovie: PropTypes.func.isRequired,
+  movie: PropTypes.object.isRequired,
+};
+
+//TODO: implement in Routing module?
+MovieDetails.defaultProps = {
+  movieId: 269149,
+};
+
+const mapStateToProps = state => ({
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/MovieDetails/MovieDetails.js]: please use redux hooks instead 

https://react-redux.js.org/api/hooks

File: netflix-roulette/src/components/actions.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import * as constants from './constants';
+
+const getMoviesSuccessAction = movies => 
+  ({ type: constants.GET_MOVIES, payload: movies });
+
+const getMovieSuccessAction = movie => 
+  ({ type: constants.GET_MOVIE, payload: movie });
+
+export const getMoviesAction = (filter = undefined, sortBy = constants.DEFAULT_SORT_BY) => async (dispatch) => {
+  let url = `${constants.URL}?sortBy=${sortBy}&sortOrder=${constants.DEFAULT_SORT_ORDER}&limit=${constants.DEFAULT_LIMIT}`;
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]:
```
getMoviesSuccessAction
getMoviesErrorAction
getMoviesPendingAction
```

Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]: dispatch(getMoviesPendingAction())

File: netflix-roulette/src/components/actions.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  if (filter) {
+    url = `${url}&filter=${filter}`;
+  }
+  const response = await fetch(url).then(res => res.json());
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]: const response = await fetch(url).then(res => dispatch(getMoviesSuccessAction(res.json())));

File: netflix-roulette/src/components/store.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+
+import rootReducer from './rootReducer';
+
+const Store = () => {
+  const initialState = {
+    movies: [],
+    movie: { },
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/store.js:9]: isLoading: false,

File: netflix-roulette/src/components/rootReducer.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import * as constants from './constants';
+
+const reducer = (state = {}, action) => {
+  switch (action.type) {
+    case constants.GET_MOVIES: {
+      return {...state, movies: action.payload };
+    }
+    case constants.GET_MOVIE: {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/rootReducer.js]: case constants.GET_MOVIES_PENDING: {
      return {...state, isLoading: true };
    }

case constants.GET_MOVIES_SUCCESS {
      return {...state, isLoading: false, movies: action.payload };
    }

case constants.GET_MOVIES_ERROR {
      return {...state, isLoading: false, error: action.payload };
    }

File: netflix-roulette/src/components/actions.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+};
+
+export const createMovieAction = (input) => async (dispatch) => {
+  const response = await fetch(`${constants.URL}`, {
+      body: input,
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+      },
+    });
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]: .then(dispatch(getMoviesAction()))

File: netflix-roulette/src/components/actions.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  const response = await fetch(url).then(res => res.json());
+  dispatch(getMoviesSuccessAction(response.data));
+};
+
+export const getMovieAction = (id) => async (dispatch) => {
+   const response = await fetch(`${constants.URL}/${id}`).then(res => res.json());
+   dispatch(getMovieSuccessAction(response));
+};
+
+export const createMovieAction = (input) => async (dispatch) => {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/actions.js]: `async (dispatch) =>`. - --  `async (dispatch, getState) =>`.  

getState() -->

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js [hunk 9]
```diff
@@ -1,21 +1,26 @@
...
-const MovieDetails = () => {
-  const movie = data[3];
+const MovieDetails = ({ getMovie, movieId, movie }) => {
   const duration = `${movie.runtime} ${wordings.min}`;
   const year = movie ? new Date(movie.release_date).getFullYear() : '-';
 
+  useEffect(() => {
+    getMovie(movieId);
+  }, [])
+
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/MovieDetails/MovieDetails.js:33]: `const {movies} = useSelector((state) => state)`

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js [hunk 10]
```diff
@@ -1,21 +1,26 @@
...
 import Header from '/src/components/shared/Header/Header';
 import StoryCard from '/src/components/shared/StoryCard/StoryCard';
 import MoviesList from '/src/components/shared/MoviesList/MoviesList';
+import { getMovieAction } from '/src/components/actions';
 
 import styles from './MovieDetails.module.scss';
 
-const MovieDetails = () => {
-  const movie = data[3];
+const MovieDetails = ({ getMovie, movieId, movie }) => {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/MovieDetails/MovieDetails.js]: const dispatch = useDispatch()

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js [hunk 11]
```diff
@@ -1,21 +1,26 @@
...
   const duration = `${movie.runtime} ${wordings.min}`;
   const year = movie ? new Date(movie.release_date).getFullYear() : '-';
 
+  useEffect(() => {
```
Comment (omelenetskiy) [inline netflix-roulette/src/components/features/MovieDetails/MovieDetails.js:27]: dispatch(getMovieAction())

File: netflix-roulette/src/components/actions.js (hunk 7 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/actions.js]: done.

File: netflix-roulette/src/components/actions.js (hunk 1 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/actions.js]: await fetch(url).then((res) => res.json()).then((res) => dispatch(getMoviesSuccessAction(res.data)));

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js (hunk 2 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/MovieDetails/MovieDetails.js]: done.

File: netflix-roulette/src/components/actions.js (hunk 3 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/actions.js]: done.

Comment (UltramarineDev) [reply netflix-roulette/src/components/actions.js]: done.

File: netflix-roulette/src/components/actions.js (hunk 4 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/actions.js]: await fetch(url).then((res) => res.json()).then((res) => dispatch(getMoviesSuccessAction(res.data)));

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js (hunk 9 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/MovieDetails/MovieDetails.js:33]: done.

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js (hunk 10 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/MovieDetails/MovieDetails.js]: done.

File: netflix-roulette/src/components/features/MovieDetails/MovieDetails.js (hunk 11 above)
Comment (UltramarineDev) [reply netflix-roulette/src/components/features/MovieDetails/MovieDetails.js:27]: done.

**Pull request merged**